agent/worker_test.go

```diff
@@ -390,6 +465,22 @@ func TestWorkerUpdate(t *testing.T) {
 					},
 					Action: api.AssignmentChange_AssignmentActionUpdate,
 				},
+				{
+					Assignment: &api.Assignment{
+						Item: &api.Assignment_Config{
+							Config: &api.Config{ID: "config-1"},
+						},
+					},
+					Action: api.AssignmentChange_AssignmentActionRemove,
+				},
+				{
+					Assignment: &api.Assignment{
+						Item: &api.Assignment_Config{
+							Config: &api.Config{ID: "config-2"},
+						},
+					},
+					Action: api.AssignmentChange_AssignmentActionUpdate,
+				},
 			},
 			expectedTasks: []*api.Task{
 				{ID: "task-1"},
```

Comment:
Unrelated to your change, and I'm probably misunderstanding how this behaves, but this particular test case has the action `api.AssignmentChange_AssignmentActionRemove` for `task-1`.   Shouldn't `task-1` no longer be in the list of expected tasks?